### PR TITLE
[Experimental] Fix depth calculation issues

### PR DESCRIPTION
**Pull request by waken22**
This is a Experimental PR to try to improve the rendering to behave better in some scenarios where sprites were colliding with 3d models. It's not perfect, and is my attempt to improve it.

Do **not** merge until we're sure it doesn't make any additional issues.

Thanks to @Danil0v3s to provide the magic formula.

### CODE REVIEW

**Comment by Danil0v3s** [comment]
Could you provide some evidences?

**Comment by waken22** [comment]
https://gyazo.com/6b1f8f326a2baff4aa510d251850c3e1
https://gyazo.com/f14a71574d751dbda3d9c60ee7f22f19
https://gyazo.com/4d2bf0952aa55f25b3b9474a3a8324bb
Test1

As i said it's not perfect though:
https://gyazo.com/f0f545fd65f752482f078333bf4c66c6
https://gyazo.com/9353ff82bbf99a173fbc0decb4668932

File: src/Renderer/SpriteRenderer.js [hunk 1]
```diff
@@ -60,26 +60,36 @@ function(      WebGL,         glMatrix,      Camera )
 			mat[3].y += mat[0].y * x + mat[1].y * y + mat[2].y * z;
 			mat[3].z += (mat[0].z * x + mat[1].z * y + mat[2].z * z) + (uCameraLatitude * floor(min(uCameraZoom, 1.0)) / 50.0);
```
Comment (Danil0v3s) [inline src/Renderer/SpriteRenderer.js:61]: Regarding my last comment, what happens if you comment this out?

File: src/Renderer/SpriteRenderer.js [hunk 2]
```diff
@@ -60,26 +60,36 @@ function(      WebGL,         glMatrix,      Camera )
...
 			mat[3].w += mat[0].w * x + mat[1].w * y + mat[2].w * z;
-			
+
 			// Spherical billboard
 			mat[0].xyz = vec3( 1.0, 0.0, 0.0 );
 			mat[1].xyz = vec3( 0.0, 1.0, 0.0 );
 			mat[2].xyz = vec3( 0.0, 0.0, 1.0 );
```
Comment (Danil0v3s) [inline src/Renderer/SpriteRenderer.js:67]: And this, whats the outcome if you comment it?

**Comment by waken22** [comment]
> By the looks of it, it seems like the billboarding is still being applied to the vertical axis of the sprite 🤔

Mmm removing the billboard make the sprite to start to distort... so i gave up and instead i used the formula to apply it to the Z.

File: src/Renderer/SpriteRenderer.js (hunk 1 above)
Comment (waken22) [reply src/Renderer/SpriteRenderer.js:61]: Changed to `			mat[3].z += (mat[0].z * x + mat[1].z * y + mat[2].z * z);`
If i comment it out entirely it just projects wrong the sprites completely

f913d58584f08221e8e27b3657db34e4
12004c054d0dc7fb2c164c8a29dbe419

File: src/Renderer/SpriteRenderer.js (hunk 2 above)
Comment (Danil0v3s) [reply src/Renderer/SpriteRenderer.js:67]: Which line did you comment? All of it?

**Pull request closed without merging**